geospaas_processing/downloaders.py

```diff
@@ -39,27 +41,104 @@ class TooManyDownloadsError(DownloadError):
 
 class Downloader():
     """Base class for downloaders"""
+
+    @staticmethod
+    def get_auth(kwargs):
+        """Builds the `auth` argument taken by `requests.get()` from
+        the keyword arguments. Uses Basic Auth.
+        """
+        if ('username' in kwargs) and ('password_env_var') in kwargs:
+            return (kwargs['username'], os.getenv(kwargs['password_env_var']))
+        return (None, None)
+
     @classmethod
-    def check_and_download_url(cls, url, download_dir, file_prefix=None, **kwargs):
+    def connect(cls, url, auth=(None, None)):
+        """Connect to the remote repository. This should return an
+        object from which the file to download can be read.
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def close_connection(cls, connection):
+        """Closes the open connection. Most objects used as connections
+        have a close() method, but it might be necessary to override
+        this in child classes.
+        """
+        connection.close()
```

Comment:
Is there any `close()` method for the response of `requests`library? I am asking this because you return it in the case HTTP. How you test the downling in the HTTP case with this PR?

Reply:
Yes, `requests.Response` has a close method and there are unit tests for HTTP downloads.